c/parallel/test/test_radix_sort.cpp

```diff
@@ -0,0 +1,213 @@
+      decomposer,
+      num_items,
+      begin_bit,
+      end_bit,
+      is_overwrite_okay,
+      selector,
+      0));
+
+  REQUIRE(CUDA_SUCCESS == cccl_device_radix_sort_cleanup(&build));
+}
+
+TEMPLATE_LIST_TEST_CASE("DeviceRadixSort::SortKeys works", "[radix_sort]", key_types)
+{
+  // We want a mix of small and large sizes because different implementations will be called
+  const int num_items                 = GENERATE_COPY(take(2, random(1, 1000000)), values({500, 1000000, 2000000}));
+  const bool is_descending            = GENERATE(false, true);
+  const auto order                    = is_descending ? CCCL_DESCENDING : CCCL_ASCENDING;
+  std::vector<TestType> input_keys    = make_shuffled_sequence<TestType>(num_items);
+  std::vector<TestType> expected_keys = input_keys;
+
+  pointer_t<TestType> input_keys_it(input_keys);
+  pointer_t<TestType> output_keys_it(num_items);
+
+  pointer_t<item_t> input_items_it, output_items_it;
+
+  int begin_bit          = 0;
+  int end_bit            = sizeof(TestType) * 8;
+  bool is_overwrite_okay = GENERATE(false, true);
+  int selector           = -1;
+
+  radix_sort(
+    order,
+    input_keys_it,
+    output_keys_it,
+    input_items_it,
+    output_items_it,
+    cccl_op_t{},
+    "",
+    num_items,
+    begin_bit,
+    end_bit,
+    is_overwrite_okay,
+    &selector);
+
+  assert(selector == 0 || selector == 1);
+
+  if (is_descending)
+  {
+    std::sort(expected_keys.begin(), expected_keys.end(), std::greater<TestType>());
+  }
+  else
+  {
+    std::sort(expected_keys.begin(), expected_keys.end());
+  }
+
+  auto& output_keys = (is_overwrite_okay && selector == 0) ? input_keys_it : output_keys_it;
+  REQUIRE(expected_keys == std::vector<TestType>(output_keys));
+}
+
+TEMPLATE_LIST_TEST_CASE("DeviceRadixSort::SortPairs works", "[radix_sort]", key_types)
+{
+  const int num_items              = GENERATE_COPY(take(2, random(1, 1000000)), values({500, 1000000, 2000000}));
+  const bool is_descending         = GENERATE(false, true);
+  const auto order                 = is_descending ? CCCL_DESCENDING : CCCL_ASCENDING;
+  std::vector<TestType> input_keys = make_shuffled_sequence<TestType>(num_items);
+  std::vector<item_t> input_items(num_items);
+  std::transform(input_keys.begin(), input_keys.end(), input_items.begin(), [](TestType key) {
+    return static_cast<item_t>(key);
+  });
+
+  std::vector<TestType> expected_keys = input_keys;
+  std::vector<item_t> expected_items  = input_items;
+
+  pointer_t<TestType> input_keys_it(input_keys);
+  pointer_t<TestType> output_keys_it(num_items);
+
+  pointer_t<item_t> input_items_it(input_items);
+  pointer_t<item_t> output_items_it(num_items);
+
+  int begin_bit          = 0;
+  int end_bit            = sizeof(TestType) * 8;
+  bool is_overwrite_okay = GENERATE(false, true);
+  int selector           = -1;
+
+  radix_sort(
+    order,
+    input_keys_it,
+    output_keys_it,
+    input_items_it,
+    output_items_it,
+    cccl_op_t{},
+    "",
```

Comment:
Just for readability I would define `cccl_op_t decomposer_no_op{};` and `constexpr const char *unused_decomposer_retty = ""` and use these variables in the call to `radix_sort`.